src/libraries/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -1769,8 +1772,8 @@ static bool INumberBase<decimal>.TryConvertToSaturating<TOther>(decimal value, [
             }
             else if (typeof(TOther) == typeof(Half))
             {
-                Half actualResult = (value >= +65520.0m) ? Half.MaxValue :
-                                    (value <= -65520.0m) ? Half.MinValue : (Half)value;
+                Half actualResult = (value >= +65520.0m) ? Half.PositiveInfinity :
```

Comment:
I'm assuming this is how Single and Double saturate too? Using infinity instead of MaxValue?